### PR TITLE
ENH: Add github workflow for smoke testing DeepSMILEPandaSlidesBenchmark

### PR DESCRIPTION
**Pull request by mebristo**
Add a github workflow for running the DeepSMILEPandaSlidesBenchmark smoke test on AML. This test runs in parallel to the rest of the checks and takes roughly 6 minutes to complete once running in AML so it will not increase the workflow time noticeably, provided that enough compute is available on AML.

### CODE REVIEW

File: hi-ml-cpath/Makefile [hunk 1]
```diff
@@ -120,6 +120,17 @@ smoke_test_slidespandaimagenetmil:
 	--max_bag_size=3 --max_bag_size_inf=3 --max_epochs=2;\
 	}
 
+# Once running in AML the following test takes around 6 minutes
+smoke_test_slidespandaimagenetmil_aml:
+	{ \
+	cd ../ ;\
+	python hi-ml/src/health_ml/runner.py --model=health_cpath.SlidesPandaImageNetMILBenchmark \
+	--cluster=testing-nc24x2 --mount_in_azureml --conda_env hi-ml-cpath/environment.yml --is_finetune \
+	--conda_env hi-ml-cpath/environment.yml --crossval_count=0 --num_top_slides=2 --num_top_tiles=2 \
```
Comment (kenza-bouzid) [inline hi-ml-cpath/Makefile:129]: ` --conda_env hi-ml-cpath/environment.yml ` is duplicated

File: .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  HIML_RESOURCE_GROUP: ${{ secrets.HIML_RESOURCE_GROUP }}
+  HIML_SUBSCRIPTION_ID: ${{ secrets.HIML_SUBSCRIPTION_ID }}
+  HIML_WORKSPACE_NAME: ${{ secrets.HIML_WORKSPACE_NAME }}
+  HIML_SERVICE_PRINCIPAL_ID: ${{ secrets.HIML_SERVICE_PRINCIPAL_ID }}
+  HIML_SERVICE_PRINCIPAL_PASSWORD: ${{ secrets.HIML_SERVICE_PRINCIPAL_PASSWORD }}
+
+jobs:
+
+  smoketest:
+    runs-on: ubuntu-18.04
```
Comment (kenza-bouzid) [inline .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml:29]: OS version upgraded to 20.04 in #511

File: .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Install hi-ml-azure locally
+        run: |
+          cd hi-ml-azure
+          make pip_local
+
+      - name: smoke test
+        run: |
+          cd ${{ env.folder }}
+          make smoke_test_slidespandaimagenetmil_aml
```
Comment (kenza-bouzid) [inline .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml:61]: is there a reason why this is a separate workflow and not part of cpath-pr?

**Review by kenza-bouzid — APPROVED** [review]
LGTM, only minor comments!

File: .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml (hunk 3 above)
Comment (mebristo) [reply .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml:61]: I kept it separate as I had envisaged this becoming a file of all smoke tests that we want to run and it felt like a sensible separation but maybe it does make more sense to keep it under the Pathology PR heading - thoughts?

Comment (kenza-bouzid) [reply .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml:61]: I would suggest keeping it under the Pathology PR as the current checks are module specific. Also, we might also want to add some smoke tests to hi-ml-multimodal in the future... where should these live then?

Comment (mebristo) [reply .github/workflows/smoke-test-SlidesPANDAImageNetMILBenchmark.yml]: Makes sense to me. Have moved it

**Review by ant0nsc — APPROVED** [review]
Looking good, but a question about maintainability

File: hi-ml-cpath/Makefile [hunk 4]
```diff
@@ -109,17 +109,28 @@ regression_test_crck_simclr:
...
 	cd ../ ;\
 	python hi-ml/src/health_ml/runner.py --model=health_cpath.SlidesPandaImageNetMILBenchmark \
 	--mount_in_azureml --conda_env hi-ml-cpath/environment.yml --is_finetune \
-	--conda_env hi-ml-cpath/environment.yml --crossval_count=0 --num_top_slides=2 --num_top_tiles=2 \
+	--crossval_count=0 --num_top_slides=2 --num_top_tiles=2 \
 	--pl_limit_train_batches=2 --pl_limit_val_batches=2 --pl_limit_test_batches=2 \
 	--max_bag_size=3 --max_bag_size_inf=3 --max_epochs=2;\
 	}
 
+# Once running in AML the following test takes around 6 minutes
```
[the remark this answers is not in the document]
Comment (mebristo) [reply hi-ml-cpath/Makefile:128]: Good point. I have added variables to avoid this

**Pull request merged**